### ISSUE TITLE
The auto-updated csharp language has an issue with OnigScanner

### ISSUE THREAD

**Issue by orta** [shikijs/shiki#219]
https://github.com/shikijs/shiki/pull/218/checks?check_run_id=3497151928#step:10:10

```
  ● validates all languages can show a hello-world › csharp

    105,110,118,97,108,105,100,32,112,97,116,116,101,114,110,32,105,110,32,108,111,111,107,45,98,101,104,105,110,100
     |     let result = grammar.tokenizeLine2(line, ruleStack)
     |     let tokensLength = result.tokens.length / 2
     |     for (let j = 0; j < tokensLength; j++) {

      at OnigScanner.findNextMatchSync (node_modules/onigasm/src/OnigScanner.ts:129:27)
      at node_modules/vscode-textmate/release/webpack:/vscodetextmate/out/grammar.js:586:26
      at lineTokens (node_modules/vscode-textmate/release/webpack:/vscodetextmate/out/grammar.js:537:5)
      at f (node_modules/vscode-textmate/release/webpack:/vscodetextmate/out/grammar.js:706:13)

```

**Comment by Gerrit0** [comment]
Of course, I didn't see this until now... more info:

```js
"105,110,118,97,108,105,100,32,112,97,116,116,101,114,110,32,105,110,32,108,111,111,107,45,98,101,104,105,110,100"
.split(",").map(c => String.fromCharCode(+c)).join("")
```
Gives the error, "invalid pattern in look-behind"

- Pull request to "fix" this - https://github.com/microsoft/vscode/pull/132400
- Pull request which originally introduced this - https://github.com/dotnet/csharp-tmLanguage/pull/216

**Comment by octref** [comment]
Good catch. Let's point to an old version for now to make master green. I subscribed to the upstream issue so once this is fixed I'll upgrade to latest.

**Comment by octref** [comment]
Let's track the dotnet one for latest C# grammar. Master is green again!

**Comment by srmagura** [comment]
Just wanted to point out two things, which you may already be aware of 😅 

1. This error is affecting the latest version of shiki that's published on npm (0.9.11), not just master.
2. This error also affects the `razor` language, which makes sense since Razor is based on C#.

Code that produces the error:
```js
const shiki = require("shiki");

shiki
  .getHighlighter({
    theme: "nord",
  })
  .then((highlighter) => {
    console.log(
      highlighter.codeToHtml(
        `
using (var sha256 = SHA256.Create())
{
}
`,
        "razor"
      )
    );
  });

```
